File: acode-mac/ACode/Sources/Services/ProviderEnvGenerator.swift
import Foundation

/// Provider 环境变量生成器
/// 根据激活的 Provider 生成注入到 PTY 进程的环境变量
enum ProviderEnvGenerator {

    /// 为指定 Provider 生成环境变量字典
    static func generate(for provider: Provider) -> [String: String] {
        var env: [String: String] = [:]

        switch provider.tool {
        case "claude_code":
            if !provider.apiKey.isEmpty {
                env["ANTHROPIC_API_KEY"] = provider.apiKey
            }
            if !provider.apiBase.isEmpty {
                env["ANTHROPIC_BASE_URL"] = provider.apiBase
            }
            if !provider.model.isEmpty {
                env["ANTHROPIC_MODEL"] = provider.model
            }

        case "openai":
            if !provider.apiKey.isEmpty {
                env["OPENAI_API_KEY"] = provider.apiKey
            }
            if !provider.apiBase.isEmpty {
                env["OPENAI_BASE_URL"] = provider.apiBase
            }
            if !provider.model.isEmpty {
                env["OPENAI_MODEL"] = provider.model
            }

        case "gemini":
            if !provider.apiKey.isEmpty {
                env["GOOGLE_API_KEY"] = provider.apiKey
                env["GEMINI_API_KEY"] = provider.apiKey
            }
            if !provider.apiBase.isEmpty {
                env["GOOGLE_GEMINI_BASE_URL"] = provider.apiBase
                env["GEMINI_BASE_URL"] = provider.apiBase
            }
            if !provider.model.isEmpty {
                env["GEMINI_MODEL"] = provider.model
            }

        default:
            break
        }

        // 合并 extra_env
        if let extra = provider.extraEnvDict {
            for (key, value) in extra {
                env[key] = value
            }
        }

        return env
    }
}

File: acode-windows/lib/services/provider_env_generator.dart
import '../models/provider.dart';

/// 生成注入终端的环境变量
class ProviderEnvGenerator {
  /// 根据 Provider 生成环境变量字典
  static Map<String, String> generate(Provider provider) {
    final env = <String, String>{};

    switch (provider.tool) {
      case 'claude_code':
        if (provider.apiKey.isNotEmpty) {
          env['ANTHROPIC_API_KEY'] = provider.apiKey;
        }
        if (provider.apiBase.isNotEmpty) {
          env['ANTHROPIC_BASE_URL'] = provider.apiBase;
        }
        if (provider.model.isNotEmpty) {
          env['ANTHROPIC_MODEL'] = provider.model;
        }
        break;

      case 'openai':
        if (provider.apiKey.isNotEmpty) {
          env['OPENAI_API_KEY'] = provider.apiKey;
        }
        if (provider.apiBase.isNotEmpty) {
          env['OPENAI_BASE_URL'] = provider.apiBase;
        }
        if (provider.model.isNotEmpty) {
          env['OPENAI_MODEL'] = provider.model;
        }
        break;

      case 'gemini':
        if (provider.apiKey.isNotEmpty) {
          env['GOOGLE_API_KEY'] = provider.apiKey;
          env['GEMINI_API_KEY'] = provider.apiKey;
        }
        if (provider.apiBase.isNotEmpty) {
          env['GOOGLE_GEMINI_BASE_URL'] = provider.apiBase;
          env['GEMINI_BASE_URL'] = provider.apiBase;
        }
        if (provider.model.isNotEmpty) {
          env['GEMINI_MODEL'] = provider.model;
        }
        break;
    }

    // 合并 extraEnv
    final extra = provider.extraEnvDict;
    if (extra != null) {
      env.addAll(extra);
    }

    return env;
  }

  /// 合并多个工具的环境变量
  static Map<String, String> mergeAll(Map<String, Provider> activeProviders) {
    final env = <String, String>{};
    for (final provider in activeProviders.values) {
      env.addAll(generate(provider));
    }
    return env;
  }
}
